Projektskabelon-master/src/main/java/FunctionLayer/OverlayMaterialCalculator.java

```diff
@@ -204,13 +210,15 @@ public  static Material screwForOverlayOneWall ( Wall wall, String overlayName)
         greb.setComment("Greb til skurdør");
         greb.setSize(1);
         greb.setAmount(1);
+        greb.setPrice(LogicFacade.getPrice(greb.getId()));
         doorMaterials.add(greb);
 
         Material hinge = new Material();
         hinge.setComment("Hængsler til skurdør");
         hinge.setName("T-Hængsel 390 mm");
         hinge.setAmount(2);
         hinge.setSize(1);
+        hinge.setPrice(LogicFacade.getPrice(hinge.getId()));
```

Comment:
If you set the price here, it will be calculated for the needed size, not for the available one. I calculate de available sizes first in function layer